pkg/cas/hardlinking_file_fetcher.go

```diff
@@ -78,7 +82,7 @@ func (ff *hardlinkingFileFetcher) GetFile(ctx context.Context, blobDigest digest
 		ff.evictionSet.Touch(key)
```

Comment:
About 10 lines up from here, you can see that we generate a key for every object: `${sha}-${size}${isExecutable}`.

When doing cloning, we don't need to take `isExecutable` into account. Every clone can have its own permissions, meaning the cache doesn't need to hold up to two copies of every file. Do we want to make use of that property, or do we find it unlikely that this is going to reduce disk space in any measurable way? If it's the latter, can we add a comment up there that states this? Thanks!